core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1930,6 +1930,12 @@ class SparkContext(config: SparkConf) extends Logging {
     Utils.tryLogNonFatalError {
       _executorAllocationManager.foreach(_.stop())
     }
+    if (_dagScheduler != null) {
```

Comment:
why this change?

Reply:
This is to fix https://github.com/apache/spark/pull/21898#issuecomment-410499090 , previously LiveListenerBus was stopped before we stop DAGScheduler.

Reply:
Add a comment